### PR TITLE
22295 Allow user define what is failure in terms on circuit breaker

### PR DESCRIPTION
**Pull request by patriknw**
Rebase of https://github.com/akka/akka/pull/22298

Refs #22295

### CODE REVIEW

File: akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala [hunk 1]
```diff
@@ -683,7 +799,8 @@ class CircuitBreaker(
      * @param body Implementation of the call that needs protected
      * @return Future containing result of protected call
      */
-    override def invoke[T](body: ⇒ Future[T]): Future[T] = callThrough(body)
+    override def invoke[T](body: ⇒ Future[T], defineFailureFn: Try[T] ⇒ Boolean): Future[T] =
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala]: Oh, so it can even transform a success into circuitbreaker failure, nice!

**Review by johanandren — APPROVED** [review]
LGTM, but, no docs?

**Comment by buaya91** [comment]
@johanandren You're right, I left docs out, created an issue to track this, might do it in the weekend when I am free

https://github.com/akka/akka/issues/22596

@patriknw Thanks for merging ^_^

**Comment by johanandren** [comment]
Great, thanks for that extra doc work/ticket @buaya91 !

**Pull request merged**